tests/query/processors/test_filter_in_select_optimizer.py

```diff
@@ -0,0 +1,156 @@
+import pytest
+
+from snuba.datasets.factory import get_dataset
+from snuba.query.expressions import Column, Literal, SubscriptableReference
+from snuba.query.logical import Query
+from snuba.query.mql.parser import parse_mql_query
+from snuba.query.processors.logical.filter_in_select_optimizer import (
+    FilterInSelectOptimizer,
+)
+
+""" CONFIG STUFF THAT DOESNT MATTER MUCH """
+generic_metrics = get_dataset(
+    "generic_metrics",
+)
+mql_context = {
+    "entity": "generic_metrics_distributions",
+    "start": "2023-11-23T18:30:00",
+    "end": "2023-11-23T22:30:00",
+    "rollup": {
+        "granularity": 60,
+        "interval": 60,
+        "with_totals": "False",
+        "orderby": None,
+    },
+    "scope": {
+        "org_ids": [1],
+        "project_ids": [11],
+        "use_case_id": "transactions",
+    },
+    "indexer_mappings": {
+        "d:transactions/duration@millisecond": 123456,
+        "d:transactions/duration@second": 123457,
+        "status_code": 222222,
+        "transaction": 333333,
+    },
+    "limit": None,
+    "offset": None,
+}
+
+""" IMPORTANT STUFF """
+optimizer = FilterInSelectOptimizer()
+mql_test_cases: list[tuple[str, dict]] = []
```

Comment:
Shouldn't this have something in it?

Reply:
sorry it actually gets initialized on line 67, ill move it up to make less confusing